core/include/thread.h

```diff
@@ -11,6 +11,96 @@
  * @ingroup     core
  * @brief       Support for multi-threading
  *
+ * Priorities
+ * ==========
+ *
+ * As RIOT is using a fixed priority @ref core_sched "scheduling algorithm",
+ * threads are scheduled based on their priority. The priority is fixed for
+ * every thread and specified during the threads creation by the `priority`
+ * parameter.
+ *
+ * The lower the priority number, the higher the priority of the thread,
+ * with 0 being the highest possible priority.
+ *
+ * The lowest possible priority is @ref THREAD_PRIORITY_IDLE - 1. The exact value
+ * depends on the platforms architecture, e.g. it would be 30 in 32-bit systems,
+ * and 14 in 16-bit systems.
+ *
+ * @note Assigning the same priority to two or more threads is usually not a
+ *       good idea. A thread in RIOT may run until it yields (@ref
+ *       thread_yield) or another thread with higher priority is runnable (@ref
+ *       STATUS_ON_RUNQUEUE) again. Having multiple threads with the same
+ *       priority may make it difficult to determine when which of them gets
+ *       scheduled and how much CPU time they will get. In most applications,
+ *       the number of threads in application is significantly smaller than the
+ *       number of available priorities, so assigning distinct priorities per
+ *       thread should not be a problem. Only assign the same priority to
+ *       multiple threads if you know what you are doing!
+ *
+ * Thread Behavior
+ * ===============
+ * In addition to the priority, flags can be used when creating a thread to
+ * alter the thread's behavior after creation. The following flags are available:
+ *
+ *  Flags                         | Description
+ *  ----------------------------- | --------------------------------------------------
+ *  @ref THREAD_CREATE_SLEEPING   | the thread will sleep until woken up manually
+ *  @ref THREAD_CREATE_WOUT_YIELD | the thread will not run immediately after creation
```

Comment:
Actually, it might if sth else (e.g., an ISR) happens that triggers a context switch. The flag is not more than an optimization and cannot be relied upon to not run the new thread before thread_create() returns.